src/api/glue-api.js

```diff
@@ -0,0 +1,35 @@
+import axios from 'axios'
+// import qs from 'qs'
+
+const ROOT_URL = 'https://glue-api.zapimoveis.com.br/v2/listings?categoryPage=RESULT'
+const villageHouse = '&unitSubTypes=VILLAGE_HOUSE&unitTypes=HOME&usageTypes=RESIDENTIAL&unitTypesV3=VILLAGE_HOUSE'
+// &business=SALE
+// &listingType=USED
+// &portal=ZAP
+// &text=Casa+de+Vila
+// &addressCountry=
+// &addressState=
+// &addressCity=
+// &addressZone=
+// &addressNeighborhood=
+// &addressStreet=
+// &addressAccounts=
+// &addressType=
+// &addressLocationId=
+// &addressPointLat=
+// &addressPointLon=
+// &addressUrl=
+// &size=24
+// &from=0
+
+export async function getAdvertisers(searchTerm) {
+  // const searchTermStringified = qs.stringify(searchTerm)
+  const searchTermStringified = new URLSearchParams(searchTerm)
+  const response = await axios
+    .get(`${ROOT_URL}${villageHouse}${searchTermStringified}`, {
+      headers: {
+        'x-domain': 'www.zapimoveis.com.br',
+      },
+    }) // return a promise
```

Comment:
What do you think about move this logic for other specialist function?

```javascript
// ./src/helpers/request-helper.js
import axios from 'axios'

/**
 * Performs a HTTP GET request and returns a promise that will be resolved by function's clients
 * @param {string} url - The URL to be used on request
 * @param { Object } headers - The header used on request
 * @returns { Promise }
 */
export function get(url, headers = {}) {
    return axios.get(url, {
      headers: {
        'x-domain': 'www.zapimoveis.com.br',
        ...headers,
      },
    })
}
```
Using:

```javascript
// ./src/api/glue-api.js
...
export async function getListings(searchTerm) {
    const searchTermStringified = new URLSearchParams(searchTerm)
    const response = await get(`${ROOT_URL}${villageHouse}${searchTermStringified}`)
    return response.data.search.result.listings
}
...
```

Reply:
I liked it.
This way it separates each responsibility.